dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -46,7 +62,6 @@ def all
   def connect_provider
     return head(:bad_request) unless can_connect_provider?
 
-    auth_hash = request.env['omniauth.auth']
```

Comment:
was this deleted purposefully? it looks like we still need `auth_hash` to set provider on line 50

Reply:
I defined some helpers for this later in the file, since we do it so often:

https://github.com/code-dot-org/code-dot-org/blob/eb4e682596fbfb754273c7b137c292a855cf6a6a/dashboard/app/controllers/omniauth_callbacks_controller.rb#L233-L239